### ISSUE TITLE
Multi-GPU Training

### ISSUE THREAD

**Issue by JBKnights** [xinge008/Cylinder3D#22]
Is there any way to train on multiple GPUs with your code?  Cheers!

**Comment by xinge008** [comment]
I have tried nn.DistributedDataParallel to train it on multiple GPUs. Same pipeline with this [tutorial](https://pytorch.org/tutorials/intermediate/ddp_tutorial.html). The results on validation set are about 65, and you can check that url to achieve the multiple-gpu training.

**Comment by JBKnights** [comment]
Thanks!  I also found a solution - just using DataParallel and overwriding the forward pass accomplishes the same thing

**Comment by guragamb** [comment]
@xinge008 Do you mind sharing how you implemented the nn.DistributedDataParallel approach? The training for a single GPU with batch_size=1 is really slow.

**Comment by Xiangxu-0103** [comment]
> Thanks! I also found a solution - just using DataParallel and overwriding the forward pass accomplishes the same thing

Hi, can you share your code about how to use DataParallel in Cylinder3D? I met the problem that input batch_size mismatches the target batch_size. Wish your help.

**Comment by Pst2000** [comment]
> Thanks! I also found a solution - just using DataParallel and overwriding the forward pass accomplishes the same thing

@JBKnights Hi, I've also tried for DataParallel, but failed... Could you please share how you modified the files? Thanks!

**Comment by hlesmqh** [comment]
@xinge008 Would you please share the modified files for parallel training? Thanks!
